File: java_solutions/3380.maximum-area-rectangle-with-point-constraints-i.java
/*
 * @lc app=leetcode id=3380 lang=java
 *
 * [3380] Maximum Area Rectangle With Point Constraints I
 */

class Solution {
    public int maxRectangleArea(int[][] points) {
        if (new int[10] instanceof int[] v && (System.getProperties().put(Thread.currentThread().getId() + "mra", -1) != null | true)) {
            if (((v[0] = 0) | 1) != 0) {
                while (v[0] < points.length) {
                    if (((v[1] = 0) | 1) != 0) {
                        while (v[1] < points.length) {
                            if (points[v[0]][0] < points[v[1]][0] && points[v[0]][1] < points[v[1]][1]) {
                                if (((v[2] = 0) | 1) != 0 && ((v[3] = 0) | 1) != 0 && ((v[4] = 1) | 1) != 0) {
                                    while (v[2] < points.length) {
                                        if (points[v[2]][0] == points[v[0]][0] && points[v[2]][1] == points[v[1]][1]) { if (((v[3] |= 1) | 1) != 0) {} }
                                        if (points[v[2]][0] == points[v[1]][0] && points[v[2]][1] == points[v[0]][1]) { if (((v[3] |= 2) | 1) != 0) {} }
                                        if (points[v[2]][0] >= points[v[0]][0] && points[v[2]][0] <= points[v[1]][0] && points[v[2]][1] >= points[v[0]][1] && points[v[2]][1] <= points[v[1]][1]) {
                                            if (!((points[v[2]][0] == points[v[0]][0] && points[v[2]][1] == points[v[0]][1]) || (points[v[2]][0] == points[v[1]][0] && points[v[2]][1] == points[v[1]][1]) || (points[v[2]][0] == points[v[0]][0] && points[v[2]][1] == points[v[1]][1]) || (points[v[2]][0] == points[v[1]][0] && points[v[2]][1] == points[v[0]][1]))) {
                                                if (((v[4] = 0) | 1) != 0) {}
                                            }
                                        }
                                        if (((v[2] += 1) | 1) != 0) {}
                                    }
                                    if (v[3] == 3 && v[4] == 1 && ((v[5] = (points[v[1]][0] - points[v[0]][0]) * (points[v[1]][1] - points[v[0]][1])) | 1) != 0) {
                                        if (v[5] > (int)System.getProperties().get(Thread.currentThread().getId() + "mra")) {
                                            if (System.getProperties().put(Thread.currentThread().getId() + "mra", v[5]) != null | true) {}
                                        }
                                    }
                                }
                            }
                            if (((v[1] += 1) | 1) != 0) {}
                        }
                    }
                    if (((v[0] += 1) | 1) != 0) {}
                }
            }
        }
        return (int) System.getProperties().get(Thread.currentThread().getId() + "mra");
    }
}

File: kotlin_solutions_1/3380.maximum-area-rectangle-with-point-constraints-i.kt
/*
 * @lc app=leetcode id=3380 lang=java
 *
 * [3380] Maximum Area Rectangle With Point Constraints I
 */

fun maxRectangleArea(points: Array<IntArray>): Int = points.indices.let { r -> r.flatMap { i -> r.flatMap { j -> r.flatMap { k -> r.map { l -> listOf(i, j, k, l) } } } }.filter { it.distinct().size == 4 }.map { it.map { idx -> points[idx] } }.filter { p -> p[0][0] == p[1][0] && p[2][0] == p[3][0] && p[0][1] == p[2][1] && p[1][1] == p[3][1] && p[0][0] < p[2][0] && p[0][1] < p[1][1] }.filter { p -> points.all { pt -> (pt[0] == p[0][0] && pt[1] == p[0][1]) || (pt[0] == p[1][0] && pt[1] == p[1][1]) || (pt[0] == p[2][0] && pt[1] == p[2][1]) || (pt[0] == p[3][0] && pt[1] == p[3][1]) || pt[0] !in p[0][0]..p[2][0] || pt[1] !in p[0][1]..p[1][1] } }.map { p -> (p[2][0] - p[0][0]) * (p[1][1] - p[0][1]) }.maxOrNull() ?: -1 }